tests/test_s3.py

```diff
@@ -15,53 +17,103 @@ def test_s3_archive_file_in_bucket(mocked_s3, s3_client):
     )
 
 
-def test_s3_get_files_iter_with_matching_csv(mocked_s3, s3_client):
+def test_s3_client_files_iter_success(mocked_s3, s3_client):
     s3_client.put_file(
-        file_content="test1,test2,test3,test4",
+        file_content="",
         bucket="dsc",
-        key="test.csv",
+        key="metadata.csv",
+    )
+
+    assert list(s3_client.files_iter(bucket="dsc")) == ["metadata.csv"]
+
+
+def test_s3_client_files_iter_with_prefix_success(mocked_s3, s3_client):
+    s3_client.put_file(
+        file_content="",
+        bucket="dsc",
+        key="workflow/folder/metadata.csv",
     )
+
+    assert list(s3_client.files_iter(bucket="dsc", prefix="workflow/folder")) == [
+        "workflow/folder/metadata.csv"
+    ]
+
+
+def test_s3_client_files_iter_with_item_identifier_success(mocked_s3, s3_client):
+    s3_client.put_file(file_content="", bucket="dsc", key="workflow/folder/aaa.pdf")
+    s3_client.put_file(file_content="", bucket="dsc", key="workflow/folder/bbb.pdf")
+
     assert list(
-        s3_client.get_files_iter(
-            bucket="dsc", file_type="csv", excluded_key_prefix="archived"
+        s3_client.files_iter(
+            bucket="dsc", prefix="workflow/folder/", item_identifier="aaa"
         )
-    ) == ["test.csv"]
+    ) == ["workflow/folder/aaa.pdf"]
 
 
-def test_s3_get_files_iter_without_matching_csv(mocked_s3, s3_client):
+def test_s3_client_files_iter_with_file_type_success(mocked_s3, s3_client):
+    s3_client.put_file(file_content="", bucket="dsc", key="workflow/folder/aaa.pdf")
+    s3_client.put_file(file_content="", bucket="dsc", key="workflow/folder/aaa.tiff")
+    s3_client.put_file(file_content="", bucket="dsc", key="workflow/folder/bbb.pdf")
+
+    assert list(
+        s3_client.files_iter(
+            bucket="dsc",
+            prefix="workflow/folder/",
+            item_identifier="aaa",
+            file_type="pdf",
+        )
+    ) == ["workflow/folder/aaa.pdf"]
+
+
+def test_s3_client_files_iter_with_exclude_prefixes_success(mocked_s3, s3_client):
     s3_client.put_file(
-        file_content="test1,test2,test3,test4",
+        file_content="",
         bucket="dsc",
-        key="archived/test.csv",
+        key="workflow/folder/metadata.csv",
     )
-    assert (
-        list(
-            s3_client.get_files_iter(
-                bucket="dsc", file_type="csv", excluded_key_prefix="archived"
-            )
-        )
-        == []
+    s3_client.put_file(
+        file_content="",
+        bucket="dsc",
+        key="workflow/folder/aaa.pdf",
+    )
+    s3_client.archive_file_with_new_key(
+        bucket="dsc",
+        key="workflow/folder/metadata.csv",
+        archived_key_prefix="workflow/folder/archived",
     )
+
+    assert list(
+        s3_client.files_iter(
+            bucket="dsc",
+            prefix="workflow/folder",
+            exclude_prefixes=["workflow/folder/archived"],
+        )
+    ) == ["workflow/folder/aaa.pdf"]
```

Comment:
Get set of tests for `files_iter`!

Reply:
Thank you! Minor, but I made a small fix to follow the format:

```text
<workflow>/batch-aaa/<123>.pdf
```

when mocking S3 prefixes for consistency!